File: bridge/src/bridge_game.cpp
#if defined(DM_PLATFORM_HTML5)
#include "bridge_game.h"
#include "bridge.h"

int bridge::game::on(lua_State* L) {
    return makeOnCallback(L, js_bridge_game_on);
}

int bridge::game::visibilityState(lua_State* L) {
    return getString(L, js_bridge_game_visibilityState);
}

#endif

File: bridge/lib/web/bridge_game.js
let js_bridge_game = {
    js_bridge_game_on: function (handler, event_name, callback) {
        bridge.game.on(UTF8ToString(event_name), state => {
            {{{ makeDynCall('vii', 'handler') }}} (callback, packToJson(state));
        });
    },

    js_bridge_game_visibilityState: function () {
        return CStrOrNull(bridge.game.visibilityState);
    }
}

mergeInto(LibraryManager.library, js_bridge_game);